runtime-modules/content/src/lib.rs

```diff
@@ -2372,22 +2378,25 @@ decl_module! {
             commitment_params: TransferParameters<T::MemberId, BalanceOf<T>>
         ) {
             let channel = Self::ensure_channel_exists(&channel_id)?;
-            ensure_actor_authorized_to_accept_channel::<T>(origin, &actor, &channel.owner)?;
 
             if let ChannelTransferStatus::PendingTransfer(ref params) = channel.transfer_status {
-                ensure!(
-                    params.transfer_params == commitment_params,
-                    Error::<T>::InvalidChannelTransferCommitmentParams
-                );
+                ensure_actor_authorized_to_accept_channel::<T>(origin, &actor, &params.new_owner)?;
+                Self::validate_channel_transfer_acceptance(&actor, &commitment_params, params)?;
             } else {
-                return Err(Error::<T>::InvalidChannelTransferStatus.into())
+                return Err(Error::<T>::InvalidChannelTransferStatus.into());
             }
 
+            let new_owner = actor_to_channel_owner::<T>(&actor)?;
+
             //
             // == MUTATION SAFE ==
             //
 
             if let ChannelTransferStatus::PendingTransfer(params) = channel.transfer_status {
+                if !params.transfer_params.is_free_of_charge() {
```

Comment:
You should check for `PendingTransfer` before mutation barrier, and return error if not, not silently succeed without side effect like now. You are actually doing this statically above, so better have that check leave a `params: TransferParameters` as a result, as I think you are doing this to do access it again.

Reply:
Good idea! Fixed.